nomad/csi_endpoint.go

```diff
@@ -679,11 +679,21 @@ func (v *CSIVolume) controllerUnpublishVolume(vol *structs.CSIVolume, claim *str
 		return nil
 	}
 
+	state := v.srv.fsm.State()
+	ws := memdb.NewWatchSet()
+
+	plugin, err := state.CSIPluginByID(ws, vol.PluginID)
+	if err != nil || plugin == nil {
```

Comment:
If we break this into an if/elseif the error message for when plugin is `nil` can avoid having `<nil>` error as part of the error string.

Reply:
Good call.

Reply:
Done in e9329a7